File: Admin/src/model/Sale.ts
class Sale {
    public saleId: number;
    public productId: number;
    public numProduct: number;
    public startAt: string;
    public endAt: string;
    public percentSale: number;
    public numProductSold: number;

    public constructor(
        saleId: number,
        productId: number,
        numProduct: number,
        startAt: string,
        endAt: string,
        percentSale: number,
        numProductSold: number,
    ) {
        this.saleId = saleId;
        this.productId = productId;
        this.numProduct = numProduct;
        this.startAt = startAt;
        this.endAt = endAt;
        this.percentSale = percentSale;
        this.numProductSold = numProductSold;
    }
}

export default Sale;

File: Application/lib/model/sale.dart
class Sale {
  int? saleId;
  int? productId;
  int? numProduct;
  String? startAt;
  String? endAt;
  int? percentSale;
  int? numProductSold;

  Sale({
    this.saleId,
    this.productId,
    this.numProduct,
    this.startAt,
    this.endAt,
    this.percentSale,
    this.numProductSold,
  });

  Sale.fromJson(Map<String, dynamic> json) {
    saleId = json['saleId'];
    productId = json['productId'];
    numProduct = json['numProduct'];
    startAt = json['startAt'];
    endAt = json['endAt'];
    percentSale = json['percentSale'];
    numProductSold = json['numProductSold'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['saleId'] = saleId;
    data['productId'] = productId;
    data['numProduct'] = numProduct;
    data['startAt'] = startAt;
    data['endAt'] = endAt;
    data['percentSale'] = percentSale;
    data['numProductSold'] = numProductSold;
    return data;
  }
}
